src/Database/LSMTree/Internal/Unsafe.hs

```diff
@@ -1685,9 +1685,10 @@ ensureSessionsMatch (t :| ts) = do
 -------------------------------------------------------------------------------}
 
 {- |
-The amount of union debt is an upper bound on the amount of computation that
-needs to be performed before an incremental union is completed. This includes
-the cost of completing incremental unions that were part of a union's input.
+Union debt represents the amount of computation that must be performed before an incremental union is completed.
```

Comment:
I'm nitpicking but I think that union debt was defined as being upper bound because it does not exactly equal the amount of computation. Changing the definition of union debt makes this less clear. Ideas?

Reply:
I'm happy to rename the type `UnionDebtBound`, but that gets a bit messy.

Reply:
Let's keep it as is for now -- if we could scale credits internally, then maybe the upper bound thing goes away